### DISCUSSION TITLE
ch6-cloud/ch6-02-lock

### DISCUSSION THREAD

**Discussion by giscus** [chai2010/advanced-go-programming-book#586] [General]
# ch6-cloud/ch6-02-lock

Go语言高级编程

https://chai2010.cn/advanced-go-programming-book/ch6-cloud/ch6-02-lock.html

**Comment by Sherlockouo** [comment]
基于redis分布式锁的那个是不是有点问题呀。为啥会有2028 不是应该只有一个协程获取到锁并执行++操作吗？求解

**Reply by cch123** [reply]
这里写错了，是 1，
而且这个代码本身也有些问题，wg.Add 该放在 for 循环外面
